mmv1/products/dataplex/Datascan.yaml

```diff
@@ -0,0 +1,640 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Datascan'
+base_url: 'projects/{{project}}/locations/{{location}}/dataScans/{{name}}'
+self_link: 'projects/{{project}}/locations/{{location}}/dataScans/{{name}}'
+exclude_resource: true
+description: |
+  Only used to generate IAM resources
+iam_policy: !ruby/object:Api::Resource::IamPolicy
+  exclude: false
+  method_name_separator: ':'
+  parent_resource_attribute: 'data_scan'
+  fetch_iam_policy_verb: :GET
+  import_format:
+    ['projects/{{project}}/locations/{{location}}/dataScans/{{name}}', '{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dataplex_datascan_primary'
+    primary_resource_id: 'example'
+    primary_resource_name:
+      'fmt.Sprintf("tf-test-dataScan%s", context["random_suffix"])'
+    test_env_vars:
+      project_name: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'locationId'
+    immutable: true
+    required: true
+    url_param_only: true
+    description: |
+      The location where the data scan should reside.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The relative resource name of the scan, of the form: projects/{project}/locations/{locationId}/dataScans/{datascan_id}, where project refers to a project_id or project_number and locationId refers to a GCP region.
+  - !ruby/object:Api::Type::String
+    name: 'uid'
+    output: true
+    description: |
+      System generated globally unique ID for the scan. This ID will be different if the scan is deleted and re-created with the same name.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      Description of the scan.
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      User friendly display name.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: |
+      User-defined labels for the scan. A list of key->value pairs.
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    output: true
+    description: |
+      Current state of the DataScan.
+    values:
+      - :STATE_UNSPECIFIED
+      - :ACTIVE
+      - :CREATING
+      - :DELETING
+      - :ACTION_REQUIRED
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    output: true
+    description: |
+      The time when the scan was created.
+  - !ruby/object:Api::Type::String
+    name: 'updateTime'
+    output: true
+    description: |
+      The time when the scan was last updated.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'data'
+    required: true
+    immutable: true
+    description: |
+      The data source for DataScan.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'entity'
```

Comment:
can we make this required as well, please?

Reply:
In this case, we plan to additional sources pretty soon.

And then the semantic will be "one-of" entity / new-property. 

In this scenario, we shouldn't add required to entity right now, since dropping required later will be a backward-incompatible change, right?